pkg/data/auth.go

```diff
@@ -132,80 +184,75 @@ func GetUser(userId string) []byte {
+	errorToken := json.NewDecoder(resp.Body).Decode(&tokenResponse)
+    if errorToken != nil {
+        panic(errorToken)
+    }
+	defer resp.Body.Close()
 
-	return playerId
+	return tokenResponse
 }
 
 func GetAdminAccessToken() string {
-	adminAccessPath := KeycloakPath + "/auth/realms/ubivius/protocol/openid-connect/token"
+	username := os.Getenv("KEYCLOAK_ADMIN_USER")
+	password := os.Getenv("KEYCLOAK_ADMIN_PASSWORD")
 
-	data := url.Values{}
-	data.Set("client_id", "ubivius-client")
-	data.Set("grant_type", "client_credentials")
-	data.Set("client_secret", "7d109d2b-524f-4351-bfda-44ecad030eef")
-
-	req, err := http.NewRequest("POST", adminAccessPath, strings.NewReader(data.Encode()))
-	if err != nil {
-		panic(err)
-	}
-	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
-	client := &http.Client{}
-	resp, err := client.Do(req)
-	if err != nil {
-		panic(err)
+	if username == "" || password == "" {
+		log.Error(ErrorEnvVar, "Some environment variables are not available for the Keycloak connection. KEYCLOAK_ADMIN_USER, KEYCLOAK_ADMIN_PASSWORD")
+		os.Exit(1)
 	}
-	defer resp.Body.Close()
 
-	log.Info("Admintoken response:", "status", resp.Status)
-	body, _ := ioutil.ReadAll(resp.Body)
-	admin_token := ExtractValue(string(body), "access_token")
-	return admin_token
-}
+	credentials := Credentials{
+		Username: username, 
+		Password: password,
+	} 
 
-// extracts the value for a key from a JSON-formatted string
-// body - the JSON-response as a string. Usually retrieved via the request body
-// key - the key for which the value should be extracted
-// returns - the value for the given key
-func ExtractValue(body string, key string) string {
-	keystr := "\"" + key + "\":[^,;\\]}]*"
-	r, _ := regexp.Compile(keystr)
-	match := r.FindString(body)
-	keyValMatch := strings.Split(match, ":")
-	return strings.ReplaceAll(keyValMatch[1], "\"", "")
-}
+	adminToken := GetAccessToken(credentials)
 
-func ExtractPlayerData(requestBody []byte) []byte {
-	username := ExtractValue(string(requestBody), "username")
-	firstName := ExtractValue(string(requestBody), "firstName")
-	lastName := ExtractValue(string(requestBody), "lastName")
-	email := ExtractValue(string(requestBody), "email")
+	return adminToken.AccessToken
+}
 
-	values := map[string]string{"firstName": firstName, "lastName": lastName, "email": email, "username": username}
-	jsonValues, _ := json.Marshal(values)
-	return jsonValues
+func UserToKeycloakUser(user User) KeycloakUser {
+	keycloakUser := KeycloakUser{
+		Username: user.Username, 
+		FirstName: user.FirstName,
+		LastName: user.LastName,
+		Email: user.Email,
+		Enabled: true,
+	}
+	return keycloakUser
 }
 
 //Extract a specific claim for the jwt token
 func ExtractClaims(tokenString string) jwt.MapClaims {
 	claims := jwt.MapClaims{}
 	_, err := jwt.ParseWithClaims(tokenString, claims, nil)
+//	_, err := jwt.ParseWithClaims(tokenString, claims, func(token *jwt.Token) (interface{}, error) {
```

Comment:
Should this be commented or removed?

Reply:
Il y une erreur retourné puisqu'on ne valide pas le token avec la clé public. Jé avais mis un log.Info de l'erreur pour pas que le code plante, mais j'ai ajouté une tâche pour la gestion des tokens... mais on parse le token pour avoir l'ID du user, mais peut-être qu'il y aurait un autre moyen qui nécessiterais pas de validé et parser le token retourné et juste l'utiliser pour donner accès aux microservices.

Reply:
I don't know about leaving this in as it could be easily forgot. If you want to save this form of the function call, just put it in the task created to finish fixing all of this 😄